### PR TITLE
Update and remove OAuth endpoints

### LINKED ISSUE #1331: Remove deprecated authentication API endpoints and add replacements
GitHub Developer API announcement:
https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue and the issue can be assigned to you so that others don't attempt to duplicate the work.

Please check out our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) guide to get started.

Thank you!

### PR DESCRIPTION
**Pull request by sh3nan1gans**
Updates based on https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters

Fixes #1331.

### CODE REVIEW

**Comment by sh3nan1gans** [comment]
@googlebot cla signed

File: github/authorizations.go [hunk 1]
```diff
@@ -273,11 +142,15 @@ func (s *AuthorizationsService) Delete(ctx context.Context, id int64) (*Response
 //
 // The returned Authorization.User field will be populated.
 //
-// GitHub API docs: https://developer.github.com/v3/oauth_authorizations/#check-an-authorization
-func (s *AuthorizationsService) Check(ctx context.Context, clientID string, token string) (*Authorization, *Response, error) {
-	u := fmt.Sprintf("applications/%v/tokens/%v", clientID, token)
+// GitHub API docs: https://developer.github.com/v3/apps/oauth_applications/#check-a-token
```
Comment (gmlewis) [inline github/authorizations.go]: According to this URL, a custom media header is needed for this endpoint:
`application/vnd.github.doctor-strange-preview+json`

Can you please add this, and anywhere else it is needed?

There are many examples in the surrounding code for how custom media headers are added (and how they are also tested, which helps us to maintain the repo by alerting us in tests when they need to be removed).

File: github/authorizations.go [hunk 2]
```diff
@@ -301,11 +174,15 @@ func (s *AuthorizationsService) Check(ctx context.Context, clientID string, toke
 //
 // The returned Authorization.User field will be populated.
 //
-// GitHub API docs: https://developer.github.com/v3/oauth_authorizations/#reset-an-authorization
-func (s *AuthorizationsService) Reset(ctx context.Context, clientID string, token string) (*Authorization, *Response, error) {
-	u := fmt.Sprintf("applications/%v/tokens/%v", clientID, token)
+// GitHub API docs: https://developer.github.com/v3/apps/oauth_applications/#reset-a-token
```
Comment (gmlewis) [inline github/authorizations.go]: This one also requires the custom media header.

File: github/authorizations.go [hunk 3]
```diff
@@ -325,71 +202,35 @@ func (s *AuthorizationsService) Reset(ctx context.Context, clientID string, toke
 // username is the OAuth application clientID, and the password is its
 // clientSecret. Invalid tokens will return a 404 Not Found.
 //
-// GitHub API docs: https://developer.github.com/v3/oauth_authorizations/#revoke-an-authorization-for-an-application
-func (s *AuthorizationsService) Revoke(ctx context.Context, clientID string, token string) (*Response, error) {
-	u := fmt.Sprintf("applications/%v/tokens/%v", clientID, token)
+// GitHub API docs: https://developer.github.com/v3/apps/oauth_applications/#delete-an-app-token
```
Comment (gmlewis) [inline github/authorizations.go]: This one also requires the custom media header.

File: github/authorizations.go [hunk 4]
```diff
@@ -325,71 +202,35 @@ func (s *AuthorizationsService) Reset(ctx context.Context, clientID string, toke
...
-
 // DeleteGrant deletes an OAuth application grant. Deleting an application's
 // grant will also delete all OAuth tokens associated with the application for
 // the user.
 //
-// GitHub API docs: https://developer.github.com/v3/oauth_authorizations/#delete-a-grant
-func (s *AuthorizationsService) DeleteGrant(ctx context.Context, id int64) (*Response, error) {
-	u := fmt.Sprintf("applications/grants/%d", id)
-	req, err := s.client.NewRequest("DELETE", u, nil)
+// GitHub API docs: https://developer.github.com/v3/apps/oauth_applications/#delete-an-app-authorization
```
Comment (gmlewis) [inline github/authorizations.go]: https://developer.github.com/v3/apps/oauth_applications/#delete-an-app-authorization

**Comment by sh3nan1gans** [comment]
Thanks for the quick turnaround @gmlewis ! I've added the custom media headers for the 4 endpoints you pointed (thanks for catching that by the way)
Please let me know if there's anything else that needs to be covered.

File: github/authorizations.go [hunk 5]
```diff
@@ -43,6 +43,10 @@ const (
 	ScopeAdminGPGKey    Scope = "admin:gpg_key"
 )
 
+const (
+	mediaTypeOAuthAppPreview = "application/vnd.github.doctor-strange-preview+json"
```
Comment (gmlewis) [inline github/authorizations.go]: Let's please move this to `github.go` around line 138 and also add the appropriate comment for consistency with the rest of the custom media headers.  Thanks!

Comment (sh3nan1gans) [reply github/authorizations.go]: Got it. Let me know if there's anything else we need to cover.

File: github/github.go [hunk 6]
```diff
@@ -135,6 +135,9 @@ const (
 
 	// https://developer.github.com/changes/2019-10-03-multi-line-comments/
 	mediaTypeMultiLineCommentsPreview = "application/vnd.github.comfort-fade-preview+json"
+
+	// https://developer.github.com/v3/previews/#new-oauth-applications-api-endpoints
```
Comment (gmlewis) [inline github/github.go]: I'm sorry I wasn't clear before.  The URLs here are for the blog posts, so this one should be:
```
// https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api
```

Comment (sh3nan1gans) [reply github/github.go]: Got it

**Comment by nightlark** [comment]
LGTM, fwiw

**Comment by gmlewis** [comment]
Thank you, @nightlark.
Merging.

**Pull request merged**